pkg/etcd/etcd.go

```diff
@@ -338,36 +363,45 @@ func (e *ETCD) infoHandler() http.Handler {
 	})
 }
 
-func joinClient(ctx context.Context, runtime *config.ControlRuntime, peers []string) (*etcd.Client, error) {
-	tlsConfig, err := toTLSConfig(runtime)
+// remoteClient returns an etcd client connected to an existing set of remote peers
+func remoteClient(ctx context.Context, runtime *config.ControlRuntime, peers []string) (*etcd.Client, error) {
+	cfg, err := getClientConfig(ctx, runtime, peers...)
 	if err != nil {
 		return nil, err
 	}
+	return etcd.New(*cfg)
+}
 
-	cfg := etcd.Config{
-		Endpoints: peers,
-		TLS:       tlsConfig,
-		Context:   ctx,
+// localClient returns etcd client connected to the local etcd instance
+func localClient(ctx context.Context, runtime *config.ControlRuntime) (*etcd.Client, error) {
```

Comment:
I dont get the point of having a separate function for this, maybe we can check the len(peers) instead of in the previous function

Reply:
One of them connects to localhost always, the other connects to an existing cluster (that may or may not include the local node) via bootstrap info.

Reply:
Went ahead and cut it out as it's mostly duplicate